src/LibChorus/FileTypeHandlers/text/TextFileTypeHandler.cs

```diff
@@ -64,26 +66,36 @@ public static string GetRawMerge(string oursPath, string commonPath, string thei
 		{
 			//NB: surrounding with quotes didn't cut it to get past paths with spaces
 
-			return RunProcess("diff3/bin/diff3.exe", "-m " + LongToShortConverter.GetShortPath(oursPath) + " " +
-				LongToShortConverter.GetShortPath(commonPath) + " " +
-				LongToShortConverter.GetShortPath(theirPath));
+			return RunProcess(_diff3Exe, new string[] {"-m", LongToShortConverter.GetShortPath(oursPath),
+				LongToShortConverter.GetShortPath(commonPath),
+				LongToShortConverter.GetShortPath(theirPath) });
 		}
 
 		protected static string SurroundWithQuotes(string path)
 		{
 			return "\"" + path + "\"";
 		}
 
-		public static string RunProcess(string filePath, string arguments)
+		public static string RunProcess(string filePath, string[] arguments)
```

Comment:
this is a public API so it may be called from outside this project making this a breaking change, is it really required or can we just leave it as is.

Reply:
Commit 4932740a restores RunProcess to its original state and puts the quoting logic in the method that calls it.